python/samples/concepts/local_models/ollama_chat_completion.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+
+import asyncio
+
+from openai import AsyncOpenAI
+
+from semantic_kernel.connectors.ai.open_ai.services.open_ai_chat_completion import OpenAIChatCompletion
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.functions.kernel_arguments import KernelArguments
+from semantic_kernel.kernel import Kernel
+
+# This concept sample shows how to use the OpenAI connector with
+# a local model running in Ollama: https://github.com/ollama/ollama
+# A docker image is also available: https://hub.docker.com/r/ollama/ollama
+# The default model used in this sample is phi3 due to its compact size.
+# At the time of creating this sample, Ollama only provides experimental
+# compatibility with the `chat/completions` endpoint:
+# https://github.com/ollama/ollama/blob/main/docs/openai.md
+# Please follow the instructions in the Ollama repository to set up Ollama.
+
+system_message = """
+You are a chat bot. Your name is Mosscap and
+you have one goal: figure out what people need.
+Your full name, should you need to know it, is
+Splendid Speckled Mosscap. You communicate
+effectively, but you tend to answer with long
+flowery prose.
+"""
+
+kernel = Kernel()
+
+service_id = "local-gpt"
+
+openAIClient: AsyncOpenAI = AsyncOpenAI(
+    api_key="fake-key",  # This cannot be an empty string, use a fake key
+    base_url="http://localhost:11434/v1",
+)
+kernel.add_service(OpenAIChatCompletion(service_id=service_id, ai_model_id="phi3", async_client=openAIClient))
```

Comment:
We have a specific Ollama connector. Do we still need to show an example of handling an Ollama model via the OpenAI client?

Reply:
> We have a specific Ollama connector. Do we still need to show an example of handling an Ollama model via the OpenAI client?

In my opinion, the Ollama connector (python/semantic_kernel/connectors/ai/ollama) has a serious issue with not providing some basic OpenAIPromptExecutionSettings compatibility in  ollama_prompt_execution_settings.py - e.g. "temperature" becomes "options"."temperature" - which is breaking e.g. the SequentialPlanning settings from python/semantic_kernel/planners/sequential_planner/Plugins/SequentialPlanning/config.json. 
I spent some time, producing my own version of the getting-started ipynbs with connectors.ai.ollama, but had to capitulate for the planning notebooks. No way to get them working with the PromptExecutionSettings compatibiity issues of OllamaPromptExecutionSettings. So its either this "easy" workaround in the PR, or needing to change / provide some basic OpenAI compatibility in the area of OllamaPromptExecutionSettings.

Yes, I know that Ollama uses "options", but I think this difference for the most common parameters like temperature, top_p, stop, seed, and some translation of max_tokens into num_predict, as well as frequency/presence_penalty into repeat_last_n/_penalty. Or use connectors.ai.open_ai

Reply:
> to

We do because Ollama supports the OpenAI chat protocol and we have customers that would like to be able to use that.